src/main/java/org/elasticsearch/cloud/aws/AwsSigner.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cloud.aws;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.SignerFactory;
+import org.elasticsearch.ElasticsearchIllegalArgumentException;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.settings.Settings;
+
+public class AwsSigner {
+
+    private AwsSigner() {
+
+    }
+
+    /**
+     * Add a AWS API Signer if set in elasticsearch.yml file. If signer does not exist
+     * we fallback to default one provided by AWS SDK which depends on the region and the service used.
+     * @param settings elasticsearch node settings
+     * @param property specific API property such as "cloud.aws.ec2.signer" or "cloud.aws.s3.signer"
+     * @param configuration AWS Client configuration
+     * @param logger Elasticsearch Logger
+     * @throws ElasticsearchIllegalArgumentException if signer does not exist
+     */
+    public static void addSigner(Settings settings, String property,
+                                 ClientConfiguration configuration, ESLogger logger)
+        throws ElasticsearchIllegalArgumentException
+    {
+        String awsSigner = settings.get(property, settings.get("cloud.aws.signer"));
+        if (awsSigner != null) {
+            try {
+                logger.debug("using AWS API signer [{}]", awsSigner);
+                // We check this signer actually exists in AWS SDK
+                // It throws a IllegalArgumentException if not found
+                SignerFactory.getSignerByTypeAndService(awsSigner, null);
+                configuration.setSignerOverride(awsSigner);
+            } catch (IllegalArgumentException e) {
+                logger.warn("wrong signer set [{}] for [{}] or [cloud.aws.signer]", awsSigner, property);
+                throw new ElasticsearchIllegalArgumentException("wrong signer set [" + awsSigner + "] for [" +
+                        property +
+                        "] or [cloud.aws.signer]");
+            }
+        }
```

Comment:
To me, this is a utility class so no need to pass a settings, a prop and a logger. I think we should have something like this:

```java
public static void configureSigner(String signer, ClientConfiguration configuration) throws ElasticsearchIllegalArgumentException {

if (awsSigner == null) {
   throw new ElasticsearchIllegalArgumentException("... cannot have an empty signer...")
}
try {
SignerFactory.getSignerByTypeAndService(awsSigner, null);
configuration.setSignerOverride(awsSigner);
} catch (IllegalArgumentException e) {
throw.... 
}
```

and let higher level class handle the exception and log things.